### PR TITLE
Added additional sample projects.

### PR DESCRIPTION
**Pull request by MoFtZ**
Added fixed size buffer example for #597.

QUESTION: Can you think of any other sample projects we should add?

### CODE REVIEW

**Review by m4rs-mt — APPROVED** [review]

**Comment by m4rs-mt** [comment]
@MoFtZ Maybe we can add an additional sample demonstrating the use of our `Random` functionality:

- Using the high-level `Random` API to fill a buffer with uniformly distributed floats (can also include the `CuRand` features)
- Using the kernel `Random` API while accessing a data view inside a GPU kernel to generate random numbers on the fly

**Comment by MoFtZ** [comment]
@m4rs-mt I'm trying to use the `RNGView` within a kernel, but it raises an exception on the CPU accelerator inside `GetRandomProvider`. Did you have any suggestions for how to fix the issue?

**Comment by MoFtZ** [comment]
Fixed issue with `RNGView` on the CPU accelerator. On the CPU accelerator, the group size was not always the same as the warp size. And even on Cuda accelerators, my group size was 30, compared to a warp size of 32, so it also had potential issues.

Also updated the `SimpleViews` sample to show offset copying to fix https://github.com/m4rs-mt/ILGPU/issues/604.

File: Src/ILGPU/Runtime/ArrayViewExtensions.cs [hunk 1]
```diff
@@ -1209,6 +1209,40 @@ public static void MemSetToZero<TView>(this TView view)
...
+        /// <summary>
+        /// Copies from the source view into the given CPU data array using the default
+        /// stream.
+        /// </summary>
+        /// <typeparam name="T">The element type.</typeparam>
+        /// <typeparam name="TView">The view type.</typeparam>
+        /// <param name="source">The source view instance.</param>
+        /// <param name="span">The CPU data target.</param>
+        /// <remarks>This method is not supported on accelerators.</remarks>
+        [NotInsideKernel]
```
Comment (m4rs-mt) [inline Src/ILGPU/Runtime/ArrayViewExtensions.cs:1221]: nice extension methods 👍

**Review by m4rs-mt — APPROVED** [review]

**Pull request merged**